code/go/0chain.net/smartcontract/storagesc/allocation.go

```diff
@@ -1592,64 +1531,31 @@ func (sc *StorageSmartContract) finishAllocation(
 	sps []*stakePool,
 	balances chainstate.StateContextI,
 ) (err error) {
-	wps, err := alloc.getAllocationPools(sc, balances)
 	if err != nil {
-		return common.NewErrorf("allocation_extending_failed", "%v", err)
-	}
-	aps := wps.activeAllocationPools(alloc.ID, t.CreationDate)
-	if len(aps) == 0 {
-		return common.NewError("fini_alloc_failed",
-			"no allocation pools to pay min lock demand")
+		return common.NewErrorf("fini_alloc_failed", "%v", err)
 	}
 
-	aps.sortExpiry()
-	apIndex := 0
 	// we can use the i for the blobbers list above because of algorithm
 	// of the getAllocationBlobbers method; also, we can use the i in the
 	// passRates list above because of algorithm of the adjustChallenges
 	for i, d := range alloc.BlobberAllocs {
 		// min lock demand rest
-		var paid currency.Coin
-		lack := d.MinLockDemand - d.Spent
 		if d.MinLockDemand > d.Spent {
-			for apIndex < len(aps) && lack > 0 {
-				pay := lack
-				if pay > aps[apIndex].Balance {
-					pay = aps[apIndex].Balance
-				}
-				aps[apIndex].Balance -= pay
-				if aps[apIndex].Balance == 0 {
-					apIndex++
-				}
-
-				paid, err = currency.AddCoin(paid, pay)
-				if err != nil {
-					return err
-				}
-				lack, err = currency.MinusCoin(lack, pay)
-				if err != nil {
-					return err
-				}
-			}
-			if lack > 0 {
+			//for clIndex < len(clients) && lack > 0 {
+			if alloc.WritePool < d.MinLockDemand-d.Spent {
```

Comment:
as d.MinLockDemand - d.Spent is used at multiple places, we can put it in a variable `minDemandLeft`

Reply:
Probably not worth the effort as min lock demand is going to be removed in https://github.com/0chain/0chain/issues/1405, so the change will only be for a week or two.

Reply:
Probably not worth the effort as min lock demand is going to be removed in https://github.com/0chain/0chain/issues/1405, so the change will only be for a week or two.